helix-core/src/main/java/org/apache/helix/task/ThreadCountBasedTaskAssignmentCalculator.java

```diff
@@ -65,10 +66,26 @@ public ThreadCountBasedTaskAssignmentCalculator(TaskAssigner taskAssigner,
       Map<String, IdealState> idealStateMap) {
     Map<String, TaskConfig> taskMap = jobCfg.getTaskConfigMap();
     Map<String, Integer> taskIdMap = jobCtx.getTaskIdPartitionMap();
+    // Check if a gap exists in the context due to previouslys
```

Comment:
The function is exactly same as what we have in GenericTaskTaskAssignmentCalculator. Can we have two function wrapped in TaskAssignmentCalculator. So these two calculator just call it?

Reply:
Done.